packages/ui/src/app/pages/Election/components/ElectionProgressBar.tsx

```diff
@@ -0,0 +1,623 @@
+    const days = Math.floor(duration / (A_DAY / MILLISECONDS_PER_BLOCK))
+    const seconds = _totalSeconds - _minutes * 60
+    const minutes = _minutes - _hours * 60
+    const hours = _hours - days * 24
+
+    return intersperse(
+      [
+        [days.toString().padStart(2, '0'), 'day'],
+        [hours.toString().padStart(2, '0'), 'hours'],
+        [minutes.toString().padStart(2, '0'), 'minutes'],
+        [seconds.toString().padStart(2, '0'), 'seconds'],
+      ].flatMap(([amount, unit]) => (amount ? <Period key={unit} amount={amount} unit={''} tiny={false} /> : [])),
+      (index) => <Separator key={index} tiny={true} />
+    )
+  }
+
+  const defaultStaging = props.electionStage
+  const defaultVerbIndicator = 'ends in'
+  const defaultDaysIndicator = useMemo(() => convertDurationToTimeString(duration), [duration])
+
+  const [stageDescription, setStageDescription] = useState(defaultStaging)
+  const [verbIndicator, setVerbIndicator] = useState(defaultVerbIndicator)
+  const [daysIndicator, setDaysIndicator] = useState<any>(defaultDaysIndicator)
+  const [selectedToolbarStage, setSelectedToolbarStage] = useState('')
+
+  useEffect(() => {
+    if (selectedToolbarStage === '') updateDescription(selectedToolbarStage, false)
+    else updateDescription(selectedToolbarStage, true)
+  }, [defaultDaysIndicator])
+
+  let announcingProgress = 0
+  let votingProgress = 0
+  let revealingProgress = 0
+  let inactiveProgress = 0
+
+  let announcingColor = Colors.Blue[500]
+  let votingColor = Colors.Blue[500]
+  const revealingColor = Colors.Blue[500]
+  let inactiveColor = Colors.Blue[500]
+
+  let remainDays = 0
+  let announcingDays = 0
+  let votingDays = 0
+  let revealingDays = 0
+  let inactiveDays = 0
+
+  let announcingEndDay = ''
+  let votingEndDay = ''
+  let revealingEndDay = ''
+  let inactiveEndDay = ''
+
+  let announcingEndBlock = 0
+  let votingEndBlock = 0
+  let revealingEndBlock = 0
+  let inactiveEndBlock = 0
+
+  let progressBarAttr = '1fr 14fr 3fr 3fr'
+
+  const constants = useCouncilConstants()
+
+  if (
+    !isNaN(duration) &&
+    constants?.budgetRefillPeriod !== undefined &&
+    constants?.election.votingPeriod != undefined &&
+    constants?.election.revealingPeriod != undefined &&
+    constants?.idlePeriod != undefined
+  ) {
+    // calculate the days per each stage
+    announcingDays = Math.floor(constants?.announcingPeriod / constants?.budgetRefillPeriod)
+    votingDays = Math.floor(constants?.election.votingPeriod / constants?.budgetRefillPeriod)
+    revealingDays = Math.floor(constants?.election.revealingPeriod / constants?.budgetRefillPeriod)
+    inactiveDays = Math.floor(constants?.idlePeriod / constants?.budgetRefillPeriod)
+
+    progressBarAttr = `${inactiveDays > 0 ? inactiveDays : 1}fr ${announcingDays > 0 ? announcingDays : 1}fr ${
+      votingDays > 0 ? votingDays : 1
+    }fr ${revealingDays > 0 ? revealingDays : 1}fr`
+
+    if (props.electionStage === 'inactive') {
```

Comment:
In this long `if { } else if { ...` the same code is repeated with just different stages please could you try to refactor this a function which doesn't repeat itself so much. As an example you can check the code of `packages/ui/src/council/hooks/useCouncilRemainingPeriod.ts`:

Line 23: an index is being assigned to each stage (in this case inactive should probably be first instead of last).
Line 17: duration are put in an array in order to get them by index

I think that by using similar a method this long file can be made a lot shorter and maintainable